src/main/kotlin/io/easybreezy/project/application/project/command/Validation.kt

```diff
@@ -53,4 +61,49 @@ class Validation @Inject constructor(private val repository: Repository) {
             validate(WriteDescription::description).isNotBlank()
         }
     }
+
+    object NoParentCategory : Constraint {
+        override val name: String
+            get() = "No category found"
+    }
+
+    suspend fun validateCommand(command: NewCategory): List<Error> {
+        return transactionManager {
+            validate(command) {
+                validate(NewCategory::name).hasSize(2, 25)
+                validate(NewCategory::parent).isNullOrProjectCategory(command.project)
+            }
+        }
+    }
+
+    suspend fun validateCommand(command: ChangeCategory): List<Error> {
+        return transactionManager {
+            validate(command) {
+                validate(ChangeCategory::name).hasSize(0, 25)
+                validate(ChangeCategory::parent).isNullOrProjectCategory(command.project)
+            }
+        }
+    }
+
+    private fun <E> Validator<E>.Property<UUID?>.isNullOrProjectCategory(project: String) {
+        this.validate(NoParentCategory) { value ->
+            value == null || repository.isProjectCategory(value, project)
+        }
+    }
+
+    object HasIssues : Constraint {
+        override val name: String
+            get() = "There are issues with this category"
+    }
+
+    suspend fun validateCommand(command: RemoveCategory): List<Error> {
+
+        return transactionManager {
+            if (repository.hasIssues(command.categoryId)) {
```

Comment:
Yes, you can use this QO. It's shouldn't be as repository method.

Reply:
Fixed